### PR TITLE
One fixup and one commit message change -- to cherry-pick, not to merge

### PR DESCRIPTION
**Pull request by j6t**
Please consider the two top-most commits here for inclusion. Please cherry-pick the fixup! commit and consider to apply the commit message of the reword! commit to the respective commit mentioned in the subject lines.

### CODE REVIEW

File: shell/ash.c [hunk 1]
```diff
@@ -10260,7 +10260,7 @@ evalcommand(union node *cmd, int flags)
 
 	expredir(cmd->ncmd.redirect);
 	redir_stop = pushredir(cmd->ncmd.redirect);
-#ifdef BASH_XTRACEFD
+#if BASH_XTRACEFD
```
Comment (dscho) [inline shell/ash.c:10263]: This change has been applied upstream already, in a different form: https://git.busybox.net/busybox/commit/?id=f8cdc7a2bcd0a9d067f5ca7da8ce7bc9c98cf34e

**Comment by dscho** [comment]
Seems that there is no place to comment on a commit message. So here goes:

> We could just write the sub-expression as get_le32(s->temp.buf + 0)

No, this is incorrect. I tested this specifically, and it left the compiler warning unchanged.

It was discussed on the BusyBox mailing list starting with [my original patch submission](http://lists.busybox.net/pipermail/busybox/2017-August/085733.html) and it was even determined that there is a bug fix for GCC relating to this.

However, it was also determined that an easier fix (at the possible expense of a slight performance hit) is to use `get_unaligned_le32()` instead, dropping my patch altogether.

That leaves only the patch that disables the `less` applet. Would you kindly rebase, keeping only that patch? I'll gladly merge after that.

Thanks!

**Comment by j6t** [comment]
I am sorry, I'm still getting warm with a workflow. I forgot to cross-check my claim that `get_le32(s->temp.buf + 0)` would not produce a warning. My main concern about the commit message is, however, that you were talking about alignment, but GCC's warning has nothing to do with alignment; it is all about the cast. Maybe you can reword that?

I do not want to submit the third patch that disables the `less` applet, yet, because I'm not yet convinced that it is a good move.

I'll close this pull request then.

**Comment by dscho** [comment]
> I am sorry, I'm still getting warm with a workflow.

No worries. It is a complex setup, with two upstreams, one of which makes it easy to contribute.

> My main concern about the commit message is, however, that you were talking about alignment, but GCC's warning has nothing to do with alignment; it is all about the cast. Maybe you can reword that?

As I stated earlier, the BusyBox team decided to do a much smaller thing and drop the unaliased `get_le32()` completely. When that fix comes into this here repository, there will not be any commit left whose commit message I could reword :smile:

**Pull request closed without merging**